public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.tsx

```diff
@@ -138,14 +139,12 @@ export const MetricsQueryEditor = (props: Props) => {
             optional
             tooltip="Change time series legend name using Dynamic labels. See documentation for details."
           >
-            <Input
-              id={`${query.refId}-cloudwatch-metric-query-editor-label`}
-              onBlur={onRunQuery}
-              value={preparedQuery.label ?? ''}
-              onChange={(event: ChangeEvent<HTMLInputElement>) =>
-                onChange({ ...preparedQuery, label: event.target.value })
-              }
-            />
+            <DynamicLabelsField
+              width={52}
+              onRunQuery={onRunQuery}
+              label={preparedQuery.label ?? ''}
+              onChange={(label) => props.onChange({ ...query, label })}
+            ></DynamicLabelsField>
```

Comment:
**Nit:** could we use a self-closing tag here?

```suggestion
            />
```

Reply:
Yes!